### PR TITLE
2FA basic auth notification email

### PR DESCRIPTION
**Pull request by di**
This updates #13830 so that the 2FA notification email on upload to still sends, with updated content.

### CODE REVIEW

File: tests/unit/forklift/test_legacy.py [hunk 1]
```diff
@@ -2593,8 +2594,17 @@ def test_upload_succeeds_with_2fa_enabled(
             IMetricsService: metrics,
         }.get(svc)
 
-        legacy.file_upload(db_request)
+        with pytest.raises(BasicAuthTwoFactorEnabled) as excinfo:
```
Comment (ewdurbin) [inline tests/unit/forklift/test_legacy.py:2597]: probably change name of this test now

**Review by ewdurbin — APPROVED** [review]
I'm iffy on breaking this out from BasicAuth policy, but for user ergonomics in the short term. Makes sense.

Approving conditionally. Let's open a PR that reestablishes enforcement via auth policy and drops the email. Merge it in a week or two?

File: tests/unit/forklift/test_legacy.py (hunk 1 above)
Comment (di) [reply tests/unit/forklift/test_legacy.py:2597]: Eh? Is the change to `test_basic_auth_upload_fails_with_2fa_enabled` not sufficient?

**Comment by di** [comment]
> Let's open a PR that reestablishes enforcement via auth policy and drops the email. Merge it in a week or two?

Agreed this should eventually be part of the policy, but I think that should probably happen when enforcement is global and not conditional for some users/projects.

File: tests/unit/forklift/test_legacy.py (hunk 1 above)
Comment (ewdurbin) [reply tests/unit/forklift/test_legacy.py:2597]: image

I guess weird folding in UI

**Pull request merged**